BinarySearchUnknownSize.java

```diff
@@ -0,0 +1,53 @@
+// Time Complexity : O(log n) where n is the number of elements in the array, but here we don't know what n is.
+// Space Complexity :  Space complexity is the size of the array itself.
```

Comment:
The space complexity is O(1), since no additional space apart from the pointers used.
Input and Output Data Structures do not qualify as "Additional Space".